csi/node.go

```diff
@@ -140,13 +140,70 @@ func (s *OsdCsiServer) NodePublishVolume(
 			err.Error())
 	}
 
-	dlog.Infof("Voume %s mounted on %s",
+	dlog.Infof("Volume %s mounted on %s",
 		req.GetVolumeId(),
 		req.GetTargetPath())
 
 	return &csi.NodePublishVolumeResponse{}, nil
 }
 
+// NodeUnpublishVolume is a CSI API call which unmounts the volume.
+func (s *OsdCsiServer) NodeUnpublishVolume(
+	ctx context.Context,
+	req *csi.NodeUnpublishVolumeRequest,
+) (*csi.NodeUnpublishVolumeResponse, error) {
+
+	dlog.Debugf("NodeUnPublishVolume req[%#v]", req)
+
+	// Check arguments
+	if req.GetVersion() == nil {
+		return nil, status.Error(codes.InvalidArgument, "Version must be provided")
+	}
+	if len(req.GetVolumeId()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Volume id must be provided")
+	}
+	if len(req.GetTargetPath()) == 0 {
+		return nil, status.Error(codes.InvalidArgument, "Target path must be provided")
+	}
+
+	// Get volume information
+	v, err := util.VolumeFromName(s.driver, req.GetVolumeId())
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Volume id %s not found: %s",
+			req.GetVolumeId(),
+			err.Error())
+	}
+
+	// Verify target location is an existing directory
+	// See: https://github.com/container-storage-interface/spec/issues/60
+	if err = verifyTargetLocation(req.GetTargetPath()); err != nil {
+		return nil, status.Error(codes.NotFound, err.Error())
+	}
+
+	// Mount volume onto the path
+	if err = s.driver.Unmount(v.GetId(), req.GetTargetPath(), nil); err != nil {
```

Comment:
Either use v.GetId() OR req.GetVolumeId() for consistency.

Reply:
yeah, that makes sense. I changed from v.GetId() to req.GetVolumeId() in NodePublishVolume